include/os/linux/spl/sys/kmem.h

```diff
@@ -32,7 +32,7 @@
 extern int kmem_debugging(void);
 extern char *kmem_vasprintf(const char *fmt, va_list ap);
 extern char *kmem_asprintf(const char *fmt, ...);
-extern char *strdup(const char *str);
+extern char *spl_strdup(const char *str);
```

Comment:
If we need to prefix `strdup` for compatibility then we should prefix its counterpart `strfree` as well.

Renaming them `kmem_strdup()` and `kmem_strfree()` would be consistent with the existing wrappers.  As of today these kmem_* names don't appears to conflict with existing symbols on any platform.  While it does make a future collision a little bit more likely, I personally like the names a bit better and think it's worth the small future risk.